implementations/swift-swift-json-schema/Sources/BowtieJSONSchema/BowtieProcessor.swift

```diff
@@ -0,0 +1,103 @@
+import Foundation
+import JSONSchema
+
+struct BowtieProcessor {
+  enum Constants {
+    static let supportedDialects: Set<String> = [
+      JSONSchema.Dialect.draft2020_12.rawValue
+    ]
+  }
+
+  private var didStart = false
+  private var currentDialect: String?
+
+  private let encoder: JSONEncoder
+  private let decoder: JSONDecoder
+
+  init(encoder: JSONEncoder, decoder: JSONDecoder) {
+    self.encoder = encoder
+    self.decoder = decoder
+  }
+
+  mutating func handle(command: Command) throws {
+    switch command {
+    case .start(let start):
+      try handleStart(start)
+    case .dialect(let dialect):
+      try handleDialect(dialect)
+    case .run(let run):
+      try handleRun(run)
+    case .stop:
+      try handleStop()
+    }
+  }
+
+  mutating func handleStart(_ start: Start) throws {
+    log("STARTING")
+    didStart = true
+    let response = StartResponse(
+      version: start.version,
+      implementation: .init(version: jsonSchemaVersion, dialects: Constants.supportedDialects)
+    )
+    try write(response)
+  }
+
+  mutating func handleDialect(_ dialect: Dialect) throws {
+    assert(didStart)
+    log("Dialect: \(dialect.dialect)")
+    try write(DialectResponse(ok: Constants.supportedDialects.contains(dialect.dialect)))
+  }
+
+  mutating func handleRun(_ run: Run) throws {
+    assert(didStart)
+    let identifier = run.seq
+    log("Schema: \(run.testCase.schema)")
+
+    let rawTestSchema = run.testCase.schema
+    let schema: Schema
+    do {
+      schema = try Schema(
+        rawSchema: rawTestSchema,
+        context: .init(
+          dialect: .draft2020_12,
```

Comment:
Even though you only (currently) support draft 2020-12 it seems a bit more future-proof to (once you follow the above on dialect requests) use the dialect which was sent previously here rather than hard-coding 2020-12.